In [1]:
# Import Dependencies
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np

In [2]:
# Load the Size History by County csv to a DataFrame
size_history_data_path = "../cleaned_data/Electric_Vehicle_Population_Size_History_By_County_Cleaned.csv"
size_history_data = pd.read_csv(size_history_data_path)

In [3]:
size_history_data.head()

Date    County State Vehicle Primary Use  \
0  January 31 2017    Juneau    AK           Passenger   
1  January 31 2017      Dale    AL           Passenger   
2  January 31 2017   Russell    AL           Passenger   
3  January 31 2017  Maricopa    AZ           Passenger   
4  January 31 2017      Pima    AZ           Passenger   

   Battery Electric Vehicles (BEVs)  Plug-In Hybrid Electric Vehicles (PHEVs)  \
0                                 0                                         1   
1                                 0                                         1   
2                                 1                                         0   
3                                 2                                         0   
4                                 0                                         1   

   Electric Vehicle (EV) Total  Non-Electric Vehicle Total  Total Vehicles  \
0                            1                          49              50   
1                            1                          71              72   
2                            1                          53              54   
3                            2                        1774            1776   
4                            1                         439             440   

   Percent Electric Vehicles  
0                       2.00  
1                       1.39  
2                       1.85  
3                       0.11  
4                       0.23

In [27]:
# Copy and group dataframe into Counties, then summarise columns of interest
county_data_EV_count = size_history_data.groupby("County")["Electric Vehicle (EV) Total"].sum()
county_data_total_count = size_history_data.groupby("County")["Total Vehicles"].sum()
county_data_percent = (county_data_EV_count / county_data_total_count)*100
county_data_percent = county_data_percent.map("{:.2f}".format)

# Assemble columns into dataframe
county_data = pd.DataFrame(
    {"Total Electric Vehicle (EV)": county_data_EV_count,
     "Total Vehicles": county_data_total_count,
     "Percent Electric Vehicles": county_data_percent
    }
)

# Sort data by descending order
county_data_sorted_percent = county_data.sort_values("Percent Electric Vehicles", ascending=False)
county_data_sorted_count = county_data.sort_values("Total Electric Vehicle (EV)", ascending=False)

# Display data
county_data_sorted_percent

Total Electric Vehicle (EV)  Total Vehicles  \
County                                                              
Charlottesville City                           11             112   
Carson City                                    40             426   
Tipton                                         22             251   
Tuolumne                                       35             405   
Williams                                       21             252   
...                                           ...             ...   
Yakima                                      13597        13324056   
Ferry                                         414          478798   
Adams                                         804         1065065   
Pend Oreille                                  702          902785   
Garfield                                       94          136058   

                     Percent Electric Vehicles  
County                                          
Charlottesville City                      9.82  
Carson City                               9.39  
Tipton                                    8.76  
Tuolumne                                  8.64  
Williams                                  8.33  
...                                        ...  
Yakima                                    0.10  
Ferry                                     0.09  
Adams                                     0.08  
Pend Oreille                              0.08  
Garfield                                  0.07  

[234 rows x 3 columns]

In [25]:
# Copy and group dataframe into States, then summarise columns of interest
state_data_EV_count = size_history_data.groupby("State")["Electric Vehicle (EV) Total"].sum()
state_data_total_count = size_history_data.groupby("State")["Total Vehicles"].sum()
state_data_percent = (state_data_EV_count / state_data_total_count)*100
state_data_percent = state_data_percent.map("{:.2f}".format)

# Assemble columns into dataframe
state_data = pd.DataFrame(
    {"Total Electric Vehicle (EV)": state_data_EV_count,
     "Total Vehicles": state_data_total_count,
     "Percent Electric Vehicles": state_data_percent
    }
)

# Sort data by descending order
state_data_sorted_percent = state_data.sort_values("Percent Electric Vehicles", ascending=False)
state_data_sorted_count = state_data.sort_values("Total Electric Vehicle (EV)", ascending=False)

# Display data
state_data_sorted_percent

Total Electric Vehicle (EV)  Total Vehicles Percent Electric Vehicles
State                                                                       
ND                              21             252                      8.33
WI                              66            1202                      5.49
LA                              91            1933                      4.71
IN                             132            2831                      4.66
NM                             119            2669                      4.46
MI                             143            3900                      3.67
MO                             103            3421                      3.01
MT                             184            7294                      2.52
MS                              39            1630                      2.39
MN                             141            5926                      2.38
AL                             116            4938                      2.35
PA                             194            8299                      2.34
OH                              66            2955                      2.23
DE                             108            4899                      2.20
NE                              52            2675                      1.94
KY                              65            3543                      1.83
AK                              70            3876                      1.81
SC                              48            2725                      1.76
NY                             200           11446                      1.75
TN                             188           11261                      1.67
OK                             109            6726                      1.62
GA                             582           43523                      1.34
DC                             161           12143                      1.33
FL                             736           58469                      1.26
KS                              72            5774                      1.25
IL                             253           20420                      1.24
MD                             749           61914                      1.21
VA                            1567          141610                      1.11
ID                             215           21334                      1.01
OR                             674           72170                      0.93
WA                         2861305       360502912                      0.79
CA                            2972          389338                      0.76
TX                             803          106945                      0.75
HI                             189           27755                      0.68
UT                              40            6050                      0.66
NC                             303           48806                      0.62
NV                             251           41457                      0.61
CO                             279           48040                      0.58
AZ                             424           89872                      0.47

**State Findings**

Washington has the highest count of electric vehicles with 2861305 vehicles, making up 0.79 percent of their total vehicles.
North Dakota has the highest percentage of electric vehicles, with 8.33% (21 electric vehicles out of 252 total vehicles).

**County Findings**

King, Washington has the highest count of electric vehicles with 1549032 vehicles, making up 1.66 percent of their total vehicles.
Charlottesville City, Virginia has the highest percentage of electric vehicles, with 9.82% (11 electric vehicles out of 112 total vehicles).

It was noted that there is far more data available for Washington and its counties, when compared to the other states and counties. Because of this, I believe that the dataframe does not include all vehicle data for other states, and thus we should consider removing other state data from our analysis as may lead to incorrect insights.